### DISCUSSION TITLE
Example for Zephyr

### DISCUSSION THREAD

**Discussion by vChavezB** [covemountainsoftware/cpputest-for-qpcpp#3] [General]
Hello @mattheweshleman , as mentioned I am testing QP CPP and zephyr. As my state machines depend heavily on the OS layer from zephyr (mutex, timers,etc) I didnt want to mock them. I was thinking if I could use the [zephyr posix](https://docs.zephyrproject.org/latest/services/portability/posix.html) support and use your work to add unit tests :)

So far it works and was wondering if you would be interested if I add a simple example. As mentioned in #1 , there were some things I had to change to make it work.

**Comment by mattheweshleman** [comment]
Thank you so much for the PR and improvement. Nice. Merged.

Can you elaborate a bit more on "add a simple example"?  Happy to consider it.

**Reply by vChavezB** [reply]
Well firstly an example that shows how to setup a zephyr project to include this repository. 

A state machine with simple states could be added to demonstrate how to use the API from CPPUTest and this repo. A Uart parser, motor controller,etc.

**Reply by mattheweshleman** [reply]
I'm happy to consider a concrete submission. Let me know if you find shortcomings in the existing example or docs. Glad to hear someone is finding this (potentially) useful!

**Reply by vChavezB** [reply]
Just opened a pull request. About the documentation, I think so far I havent had any major issues with following along the example provided.

And yes, I find it quite useful after I heard about TDD. The typical workflow I did before was debug after developing the software but I have seen that waiting for a specific state or set of conditions with the real hardware slows down what you can do with automated testing.

**Reply by mattheweshleman** [reply]
just FYI, been hit with work/family issues. Probably be another week or two before I can look at this.

**Reply by mattheweshleman** [reply]
And then was hit with COVID for the entire month of May... ugh. Again, sorry for the long delay in responding. I left my comment in the PR itself.
